config/cmake/ConfigureChecks.cmake

```diff
@@ -86,14 +86,6 @@ else ()
   set (HDF5_FILE_LOCKING_SETTING "no")
 endif ()
 
-#-----------------------------------------------------------------------------
-#  Are we going to use HSIZE_T
-#-----------------------------------------------------------------------------
-option (HDF5_ENABLE_HSIZET "Enable datasets larger than memory" ON)
-if (HDF5_ENABLE_HSIZET)
-  set (${HDF_PREFIX}_HAVE_LARGE_HSIZET 1)
-endif ()
-
```

Comment:
As far as I can tell, this is not used anywhere in the library or elsewhere in the build system.